### PR TITLE
C++: Add convenient result constructors

### PR DESCRIPTION
**Pull request by chfast**
New constructors are added based on the usage in state transition implementation in evmone.

### CODE REVIEW

**Review by yperbasis — APPROVED** [review]

File: include/evmc/evmc.hpp [hunk 1]
```diff
@@ -346,13 +346,36 @@ class result : private evmc_result
...
+    explicit result(evmc_status_code _status_code = EVMC_INTERNAL_ERROR,
+                    int64_t _gas_left = 0) noexcept
+      : evmc_result{make_result(_status_code, _gas_left, nullptr, 0)}
+    {}
+
+    /// Creates the result of contract creation.
+    ///
+    /// @param _status_code     The status code.
+    /// @param _gas_left        The amount of gas left.
+    /// @param _create_address  The address of the possibly created account. This can be valid
```
Comment (axic) [inline include/evmc/evmc.hpp:369]: How is the address valid in creation failure?

Comment (chfast) [reply include/evmc/evmc.hpp]: Check now.

File: include/evmc/evmc.h [hunk 2]
```diff
@@ -443,12 +443,11 @@ struct evmc_result
...
-     * The address of the contract created by create instructions.
+     * The address of the possibly created contract.
      *
-     * This field has valid value only if:
-     * - it is a result of the Host method evmc_host_interface::call
-     * - and the result describes successful contract creation
-     *   (evmc_result::status_code is ::EVMC_SUCCESS).
+     * The create address may be provided even though the contract creation has failed
+     * (evmc_result::status_code is not ::EVMC_SUCCESS). This is useful in situations
+     * when the address is observable, e.g. access to it remains warm.
```
Comment (axic) [inline include/evmc/evmc.h:450]: Hmm, in what case does the address remain warm?

Comment (chfast) [reply include/evmc/evmc.h:450]: In most cases of failing contact creation, but not all. Obviously.

**Pull request merged**